src/bin/ion/commands/generate/utils.rs

```diff
@@ -216,6 +224,38 @@ impl Language for RustLanguage {
             Template::Sequence => "sequence".to_string(),
         }
     }
+
+    fn namespace_separator() -> String {
+        "::".to_string()
+    }
+
+    fn add_type_to_namespace(
+        is_nested_type: bool,
+        type_name: &String,
+        namespace: &mut Vec<String>,
+    ) {
+        // e.g. For example there is a `NestedType` inside `Foo` struct. Rust code generation also generates similar modules for the generated structs.
+        // ```rust
+        // mod foo {
+        //   struct Foo {
+        //     ...
+        //     mod nested_type {
```

Comment:
Maybe I'm misunderstanding--`mod nested_type` is within the definition of `struct Foo` here?

Reply:
Oh no, my bad. Let me modify that. It should be inside `mod foo` and not `struct Foo`.